pkg/cmd/cli/cmd/importimage.go

```diff
@@ -249,151 +249,134 @@ func (o *ImportImageOptions) waitForImport(resourceVersion string) (*imageapi.Im
 }
 
 func (o *ImportImageOptions) createImageImport() (*imageapi.ImageStream, *imageapi.ImageStreamImport, error) {
+	var isi *imageapi.ImageStreamImport
 	stream, err := o.isClient.Get(o.Name)
-	from := o.From
-	tag := o.Tag
-
+	// no stream, try creating one
 	if err != nil {
 		if !errors.IsNotFound(err) {
 			return nil, nil, err
 		}
 		if !o.Confirm {
 			return nil, nil, fmt.Errorf("no image stream named %q exists, pass --confirm to create and import", o.Name)
 		}
-		from, stream = o.newImageStream()
-
-	} else {
-		// the stream already exists
-		if len(stream.Spec.DockerImageRepository) == 0 && len(stream.Spec.Tags) == 0 {
-			return nil, nil, fmt.Errorf("image stream has not defined anything to import")
-		}
-
-		if o.All {
-			// importing the entire repository
-			from, err = o.importAll(stream)
-			if err != nil {
-				return nil, nil, err
-			}
-		} else {
-			// importing a single tag
-			from, err = o.importTag(stream)
-			if err != nil {
-				return nil, nil, err
-			}
-		}
+		stream, isi = o.newImageStream()
+		return stream, isi, nil
 	}
 
-	if len(from) == 0 {
-		// catch programmer error
-		return nil, nil, fmt.Errorf("unexpected error, from is empty")
+	// the stream already exists
+	if len(stream.Spec.DockerImageRepository) == 0 && len(stream.Spec.Tags) == 0 {
+		return nil, nil, fmt.Errorf("image stream has not defined anything to import")
```

Comment:
That is an old message, I didn't change anything here, except for moving it into different method. What about 'image stream does not have valid docker images to be imported'?